src/collectFileNames.test.ts

```diff
@@ -0,0 +1,34 @@
+import path from "node:path";
+import { describe, expect, it } from "vitest";
+
+import { collectFileNames } from "./collectFileNames.js";
+
+describe("collectFileNames", () => {
+	it("should collect files with wildcard", async () => {
```

Comment:
[Naming] Nit: having the test name include what's going on helps differentiate it from others. I always try for an `it("X when Y"` approach to make _X_ and _Y_ clear.

```suggestion
	it("should collect files with wildcard when collection succeeds", async () => {
```